ros/src/camera_feed/camera_feed/camera_feed.py

```diff
@@ -0,0 +1,49 @@
+import cv2
+
+import rclpy
+import rclpy.exceptions
+import rclpy.executors
+import rclpy.logging
+
+from rclpy.node import Node
+
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge, CvBridgeError
+
+
+class CameraFeed(Node):
+    def __init__(self):
+        super().__init__("camera_feed")
+        self.bridge = CvBridge()
+        self.cap = cv2.VideoCapture(0)
+        self.publisher = self.create_publisher(Image, "/video", 10)
+
+    # gets the camera feed and publishes it to /video topic
+    def run(self):
+        while True:
+            try:
+                ret, frame = self.cap.read()
+                if not ret:
+                    continue
+                # cv2.imshow("test", frame)
+                if cv2.waitKey(1) & 0xFF == ord("q"):
```

Comment:
I don't believe that this is necessary if you're not using `imshow`. You can just interrupt the node with Ctrl+C

Reply:
I tried that but then the overhead_cv node crashed my computer everytime I ran it so I just left it in.

Doing some research tho, I think I can rewrite the code to just have cv2.waitKey(1) by itself without the if statement.
I'll update if my computer still crashes

Reply:
Maybe because it wasn't based on a timer?

Reply:
I think its because waitKey(1) makes the frame display for 1ms, without it the frame just displays and clears as fast as possible.

https://stackoverflow.com/questions/51143458/difference-in-output-with-waitkey0-and-waitkey1

Reply:
Which is why it should be based in a timer so that it doesn't draw as much CPU as possible, I believe

Reply:
Yeah that makes sense. I think cv2.waitkey is able to accomplish the wait for 1ms process to not blow up the computer. Would you prefer to just use the ros2 timer method instead?